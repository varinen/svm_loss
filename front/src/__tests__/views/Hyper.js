import React from 'react';
import {createShallow} from '@material-ui/core/test-utils';
import Hyper from '../../views/SVMLoss/Demo/Optimize/Hyper';

describe('Hyper renders correctly', () => {
    let component, props, shallow;
    const mockUpdateHyper = jest.fn();
    beforeEach(() => {
        props = {
            hyper: {reg_c: 0.1, learning_rate:0.1, loss_type: 'ww'},
            updateHyper: mockUpdateHyper
        };
        shallow = createShallow({dive: true});
        component = shallow(<Hyper {...props}/>);
    });

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Change in a select control calls updateHyper', () => {
        const additional =  [{"type": "ACTIVATE_PLOT_UPDATE"}];
        component.find('[name="loss_type"]')
            .simulate('change', {target: {name:"loss_type", value: 'ova'}});
        expect(mockUpdateHyper).toHaveBeenCalledWith("loss_type", "ova");
    });

    it('Change in a text input control calls updateHyper', () => {
        const additional =  [{"type": "ACTIVATE_PLOT_UPDATE"}];
        component.find('[name="reg_c"]')
            .simulate('change', {target: {name:"reg_c", value: '2'}});
        expect(mockUpdateHyper).toHaveBeenCalledWith("reg_c", "2");
    });

});
