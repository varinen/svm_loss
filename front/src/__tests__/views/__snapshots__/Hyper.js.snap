// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hyper renders correctly renders correctly 1`] = `
<GridContainer
  className=""
  container={true}
  spacing={2}
>
  <GridItem
    className=""
    key="loss_type"
    xs={4}
  >
    <WithStyles(ForwardRef(FormControl))
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <WithStyles(ForwardRef(InputLabel))>
        Loss formulation
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select))
        name="loss_type"
        onChange={[Function]}
        value="ww"
      >
        <WithStyles(ForwardRef(MenuItem))
          value="ww"
        >
          Weston Watkins 1999
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          value="ova"
        >
          One-vs-All
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
  </GridItem>
  <GridItem
    className=""
    key="learning_rate"
    xs={4}
  >
    <WithStyles(ForwardRef(TextField))
      label="Learning rate"
      name="learning_rate"
      onChange={[Function]}
      step="0.1"
      type="number"
      value={0.1}
    />
  </GridItem>
  <GridItem
    className=""
    key="reg_c"
    xs={4}
  >
    <WithStyles(ForwardRef(TextField))
      label="Regularization strength"
      name="reg_c"
      onChange={[Function]}
      step="0.1"
      type="number"
      value={0.1}
    />
  </GridItem>
</GridContainer>
`;
